lib/src/specs/typedef.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value/built_value.dart';
+import 'package:meta/meta.dart';
+
+import '../base.dart';
+import '../mixins/annotations.dart';
+import '../mixins/dartdoc.dart';
+import '../visitors.dart';
+import 'expression.dart';
+
+part 'typedef.g.dart';
+
+@immutable
+abstract class TypeDef extends Object
+    with HasAnnotations, HasDartDocs
+    implements Built<TypeDef, TypeDefBuilder>, Spec {
+  factory TypeDef([void Function(TypeDefBuilder)? updates]) = _$TypeDef;
```

Comment:
It'd be great if this could support generic typdefs (`typedef Fn<T extends Foo> = T Function();`), but I understand maybe no one needs it yet.

I only bring it up as a consideration because it might be a breaking change to add support for generics, which is a painful upgrade to make to code_builder, especially when migrating clients and everything that depends on code_builder internally...

Reply:
I'm a little less familiar with this library, but that adding generics support may just be adding `BuiltList<Reference> get types` to `TypeDef`? I think it would just parameterize the `name` field, and people could manually parameterize the `Expression? definition` on the RHS.